### PR TITLE
Improve whitespace handling (specifically wrt `htmlmin:ignore`)

### PR DESCRIPTION
**Pull request by j9t**
## Release Notes

* **Bug Fixes**
  * Fixed incorrect extra spaces appearing in htmlmin:ignore blocks when collapseWhitespace is enabled, particularly across ignore-block boundaries involving HTML comments and adjacent block/inline tags.

* **Tests**
  * Added comprehensive test coverage for htmlmin:ignore whitespace handling edge cases.

* **Chores**
  * Version bumped to 6.2.4.

### CODE REVIEW

File: src/htmlminifier.js [hunk 1]
```diff
@@ -1414,6 +1424,23 @@ async function minifyHTML(value, options, partialMarkup) {
...
+                if (match) {
+                  const idx = +match[1];
+                  if (idx < ignoredMarkupChunks.length) {
+                    const content = ignoredMarkupChunks[idx];
+                    const lastTagMatch = content && RE_LAST_HTML_TAG.exec(content);
+                    if (lastTagMatch) {
+                      const isClose = lastTagMatch[1].charAt(0) === '/';
+                      const tagName = options.name(isClose ? lastTagMatch[1].slice(1) : lastTagMatch[1]);
+                      effectivePrevTag = isClose ? '/' + tagName : tagName;
+                    }
```
[the remark this answers is not in the document]
Comment (j9t) [reply src/htmlminifier.js:1441]: (Feeling free to throw Claude at this; feel free to challenge if this misses something:)

We looked at extracting a shared `getBoundaryToken` helper but decided to leave the two sites as-is.
                                                                                                                                                                
The reason: the two sites intentionally use different heuristics for the previous content. `commentFinalize` (site 2) uses the first tag in the prev chunk to decide whether to collapse whitespace. For multi-element ignored content like `<p>x</p><a>y</a>`, first-tag yields `p` (block → collapse), which is correct. Switching to last-tag there would yield `/a` (inline → preserve), which would reintroduce the spurious space. A shared helper would either need to expose both strategies or silently make site 2 wrong, so we kept them separate with the `RE_LAST_HTML_TAG` constant hoisted at the top of the file for reuse.
